src/Silex/Provider/TwigServiceProvider.php

```diff
@@ -154,7 +154,20 @@ public function register(Container $app)
         };
 
         $app['twig.loader.filesystem'] = function ($app) {
-            return new \Twig_Loader_Filesystem($app['twig.path']);
+            if (!is_array($app['twig.path'])) {
+                $app['twig.path'] = array($app['twig.path']);
+            }
+
+            $fileSystem = new \Twig_Loader_Filesystem();
+            foreach ($app['twig.path'] as $namespace => $path) {
```

Comment:
the structure you suggest here does not allow adding multiple paths for the same namespace (except for the main one), which means support is only partial